### PR TITLE
:warning: check return err for customized agent install namespace

### PR DESCRIPTION
**Pull request by zhujian7**
## Summary

This PR changes the `RegistrationOption` interface:
from
```
AgentInstallNamespace func(addon *addonapiv1alpha1.ManagedClusterAddOn) string
```
to
```
AgentInstallNamespace func(addon *addonapiv1alpha1.ManagedClusterAddOn) (string, error)
```
we need this error info to determine whether the namespace is empty or an error occurred.

## Related issue(s)

Fixes #

### CODE REVIEW

File: examples/deploy/addon/helloworld-helm/resources/helloworld_helm_clustermanagementaddon.yaml [hunk 1]
```diff
@@ -9,4 +9,7 @@ spec:
   supportedConfigs:
   - group: addon.open-cluster-management.io
     resource: addondeploymentconfigs
+    defaultConfig:
```
Comment (qiujian16) [inline examples/deploy/addon/helloworld-helm/resources/helloworld_helm_clustermanagementaddon.yaml]: this is weird, if we do to specify the default configs, we should just use the default ns, right?

Comment (zhujian7) [reply examples/deploy/addon/helloworld-helm/resources/helloworld_helm_clustermanagementaddon.yaml]: If we do not specify the default config, we must specify it in every e2e case.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**